### PR TITLE
Fix consistent service naming and add test coverage

### PR DESCRIPTION
**Pull request by hlbarber**
## Motivation and Context

The casing scheme is inconsistent between the service in `lib.rs` and `service.rs` causing a bug when service names contain acronyms.

## Description

- Make service casing consistent between `lib.rs` and `service.rs`.
- Add test coverage for unusual casing.

## Notes

There is a wider problem of the server side symbol not providing a value for service. https://github.com/awslabs/smithy-rs/issues/1724

### CODE REVIEW

**Review by LukeMathWalker — APPROVED** [review]

File: codegen-core/common-test-models/naming-obstacle-course-casing.smithy [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+use aws.protocols#awsJson1_1
+
+// TODO(https://github.com/awslabs/smithy-rs/issues/2340): The commented part of the model breaks the code in a miriad
+// of ways. The solution to the linked issue must solve them.
+
+/// Confounds model generation machinery with lots of problematic casing
+@awsJson1_1
+service ACRONYMInside_Service {
+    operations: [
+        DoNothing,
```
Comment (hlbarber) [inline codegen-core/common-test-models/naming-obstacle-course-casing.smithy:13]: We don't need this operation if https://github.com/awslabs/smithy-rs/pull/2351 is merged

**Review by LukeMathWalker — APPROVED** [review]

File: codegen-core/common-test-models/naming-obstacle-course-casing.smithy [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+namespace casing
+
+use aws.protocols#awsJson1_1
+
+// TODO(https://github.com/awslabs/smithy-rs/issues/2340): The commented part of the model breaks the generator in a
+// miriad of ways. Any solution to the linked issue must address this.
+
+/// Confounds model generation machinery with lots of problematic casing
+@awsJson1_1
+service ACRONYMInside_Service {
```
Comment (crisidev) [inline codegen-core/common-test-models/naming-obstacle-course-casing.smithy:11]: I am not in love with the naming here :)

**Review by crisidev — APPROVED** [review]

File: codegen-server-test/build.gradle.kts [hunk 3]
```diff
@@ -38,6 +38,7 @@ dependencies {
 val allCodegenTests = "../codegen-core/common-test-models".let { commonModels ->
     listOf(
         CodegenTest("crate#Config", "naming_test_ops", imports = listOf("$commonModels/naming-obstacle-course-ops.smithy")),
+        CodegenTest("casing#ACRONYMInside_Service", "naming_test_casing", imports = listOf("$commonModels/naming-obstacle-course-casing.smithy")),
```
Comment (david-perez) [inline codegen-server-test/build.gradle.kts:41]: There's no harm in adding this to `codegen-client` too.

Comment (hlbarber) [reply codegen-server-test/build.gradle.kts:41]: https://github.com/awslabs/smithy-rs/pull/2349/files#diff-0e456b9761f195113cafd0d3a68068a9bd6dba3a0d4d37e8bf93d38209d36e50R82-R86

**Pull request merged**